Model/UserInterface.php

```diff
@@ -227,4 +227,74 @@ public function addRole($role);
      * @return static
      */
     public function removeRole($role);
+
+    /**
+     * Checks whether the user's account has expired.
+     *
+     * Internally, if this method returns false, the authentication system
+     * will throw an AccountExpiredException and prevent login.
+     *
+     * @return bool true if the user's account is non expired, false otherwise
+     *
+     * @see AccountExpiredException
+     */
+    public function isAccountNonExpired();
+
+    /**
+     * Checks whether the user is locked.
+     *
+     * Internally, if this method returns false, the authentication system
+     * will throw a LockedException and prevent login.
+     *
+     * @return bool true if the user is not locked, false otherwise
+     *
+     * @see LockedException
+     */
+    public function isAccountNonLocked();
+
+    /**
+     * Checks whether the user's credentials (password) has expired.
+     *
+     * Internally, if this method returns false, the authentication system
+     * will throw a CredentialsExpiredException and prevent login.
+     *
+     * @return bool true if the user's credentials are non expired, false otherwise
+     *
+     * @see CredentialsExpiredException
+     */
+    public function isCredentialsNonExpired();
+
+    /**
+     * Checks whether the user is enabled.
+     *
+     * Internally, if this method returns false, the authentication system
+     * will throw a DisabledException and prevent login.
+     *
+     * @return bool true if the user is enabled, false otherwise
+     *
+     * @see DisabledException
+     */
+    public function isEnabled();
+}
+
+// This is required to support apps that explicitly check if a user is an instance of AdvancedUserInterface
+if (interface_exists('\Symfony\Component\Security\Core\User\AdvancedUserInterface')) {
```

Comment:
This will always return true in Symfony 4.x as the class was just deprecated and not removed.

Reply:
Solution can be:
```php
use Symfony\Component\HttpKernel\Kernel;

if ( Kernel::VERSION_ID >= 40100 ){
}
```
Not sure if this is the best we can do, but it's working and nobody receive deprecation notices.

Reply:
that won't work fine, as it will detect the version of the wrong component

Reply:
@XWB probably I'm missing the point but this looks fine to me as long as this version is compliant with sf4 AND sf5.

Reply:
@DonCallisto Well, my point was that extending the `AdvancedUserInterface` will still throw a deprecation notice in Symfony 4.1 and above. I wonder how we can avoid that deprecation notice?

Reply:
That's impossible unless you create a new major version, *still compatible with sf4*, and explain that `AdvantageUserInterface` is now not available anymore (so, basically, you're introducing a BC but keeping the minimum compatibility).